app-backend/backsplash-server/src/main/scala/com/rasterfoundry/backsplash/middleware/RequestRewriteMiddleware.scala

```diff
@@ -13,35 +13,56 @@ import org.http4s.dsl.io._
 import java.util.UUID
 
 object RequestRewriteMiddleware {
-  private def getDefaultLayerId(projectId: UUID,
-                                xa: Transactor[IO]): OptionT[IO, UUID] =
+  private def getDefaultLayerId(
+      projectId: UUID,
+      xa: Transactor[IO]
+  ): OptionT[IO, UUID] =
     OptionT { ProjectDao.getProjectById(projectId).transact(xa) } map {
       _.defaultLayerId
     }
 
+  private def rewrite(
+      req: Request[IO],
+      projectId: UUID,
+      z: Int,
+      x: Int,
+      y: Int,
+      service: HttpRoutes[IO],
+      xa: Transactor[IO]
+  ) = {
+    for {
+      defaultLayerId <- getDefaultLayerId(projectId, xa)
+      resp <- service(
+        req.withPathInfo(
+          s"/${projectId}/layers/${defaultLayerId}/${z}/${x}/${y}"
+        )
+      )
+    } yield resp
+  }
+
   def apply(
       service: HttpRoutes[IO],
-      xa: Transactor[IO]): Kleisli[OptionT[IO, ?], Request[IO], Response[IO]] =
+      xa: Transactor[IO]
+  ): Kleisli[OptionT[IO, ?], Request[IO], Response[IO]] =
     Kleisli { req: Request[IO] =>
       {
         // match on the request here and route to the service if it matches the deprecated route
         val scriptName = req.scriptName
         // Due to middleware application order, we have to strip a trailing slash here
         if (req.pathInfo.charAt(req.pathInfo.length - 1) == '/') {
           apply(service, xa)(
-            req.withPathInfo(
-              req.pathInfo.substring(0, req.pathInfo.length - 1)))
+            req.withPathInfo(req.pathInfo.substring(0, req.pathInfo.length - 1))
+          )
         } else if (scriptName.isEmpty) {
           req match {
-            case GET -> Root / UUIDWrapper(projectId) / IntVar(z) / IntVar(x) / IntVar(
-                  y) =>
-              for {
-                defaultLayerId <- getDefaultLayerId(projectId, xa)
-                resp <- service(
-                  req.withPathInfo(
-                    s"/${projectId}/layers/${defaultLayerId}/${z}/${x}/${y}"))
-              } yield resp
-
+            case GET -> Root / "tiles" / UUIDWrapper(projectId) / IntVar(
```

Comment:
This is the only substantive change - anything with a prefix of `/tiles/` gets caught here and we rewrite it